virtool/samples/api.py

```diff
@@ -226,9 +226,10 @@ async def create(req):
     if name_error_message:
         return bad_request(name_error_message)
 
-    # Make sure a subtraction host was submitted and it exists.
-    if not await db.subtraction.count_documents({"_id": data["subtraction"], "is_host": True}):
-        return bad_request("Subtraction does not exist")
+    # Make sure each subtraction host was submitted and it exists.
+    for subtraction in data["subtractions"]:
```

Comment:
This can be done more efficiently with an `$in` operator and `db.subtraction.distinct()`. Compare the subtraction ID lists from the database and the API request by using the [`set`](https://docs.python.org/3/library/stdtypes.html#set) data structure.

You should return all invalid subtraction IDs in the response instead of just the first one.

Since this same check is done for analyses, you should reuse the code. Maybe add a new function: ` virtool.db.utils.check_missing_ids(id_list)`.